src/lib/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+############################################################################
```

Comment:
What changed that this file is now required? No description in PR

Reply:
Does it just build all the libs even though they might not all be used? I see modules that were really libs are now libs

Reply:
It adds (almost) all libs to the build so that the list doesn't need to repeated in every config. None of them are actually built unless pulled in by a dependency.